### ISSUE TITLE
Error when using lxrun to uninstall BashOnWindows

### ISSUE THREAD

**Issue by MaximoTrinidad** [Microsoft/WSL#1949]
This bug-tracker is monitored by developers and other technical types.  We like detail!  So please use this form and tell us, concisely but precisely, what's up.  Please fill out ALL THE FIELDS!

If you have a feature request, please post to [the UserVoice](https://wpdev.uservoice.com/forums/266908).  If you're reporting a BSOD, don't post here!  Instead, e-mail "secure@microsoft.com", and if possible attach the minidump from "C:\Windows\minidump\".

* Your Windows build number:  (Type `ver` at a Windows Command Prompt)
Microsoft Windows [Version 10.0.16176]

* What you're doing and what's happening:  (Copy&paste specific commands and their output, or include screen shots)
Error uninstalling BashOnWindows:
c:\lxrun /uninstall /full
Error: Error: 0x80070091

![bashuninstallerr_2017-04-18_18-54-42](https://cloud.githubusercontent.com/assets/19892435/25156570/6bf41190-2469-11e7-936a-2a41bdaaaadb.jpg)

It will let me install again but what guarantee that's working correctly?

* What's wrong / what should be happening instead:
It was working on older Windows 10 Insider builds.

* Strace of the failing command, if applicable:  (If `<cmd>` is failing, then run `strace -o strace.txt -ff <cmd>`, and post the strace.txt output here)

See [our contributing instructions](/CONTRIBUTING.md) for assistance.

**Comment by MaximoTrinidad** [comment]
Hum! Apparently, I haven't re build my environments in awhile as I've been upgrading my Windows 10 Insider version frequently

** The issue still valid for Windows Insider Build 16199.rs_prerelease.170513-2252 **

To repo, I  uninstall Bash using the command: lxrun /uninstall /full
And, I got the error: Error: 0x80070091

I tried to uninstall the feature from the "Control Panel" and retry to install again without any success:
PS C:\WINDOWS\system32> lxrun /install
-- Beta feature --
This will install Ubuntu on Windows, distributed by Canonical
and licensed under its terms available here:
https://aka.ms/uowterms

Type "y" to continue: y
Error: 0x80070091
PS C:\WINDOWS\system32>

Bottom line: I can't install Bash in Windows 10 Insider build any more.
:(

**Comment by benhillis** [comment]
@MaximoTrinidad - Could you collect logs while trying to uninstall / install?

https://github.com/Microsoft/BashOnWindows/blob/master/CONTRIBUTING.md#8-detailed-logs

**Comment by MaximoTrinidad** [comment]
@benhillis,

Hum!!  I just created a new Windows 10 Insider image in my Azure subscription.  
Let me see how it goes as I already did a Bash uninstall and I didn't get the 'Error: 0x80070091'.

I'm suspecting it has to be my physical machine due to all Insider's upgrades. 

I'll keep you posted!

**Comment by MaximoTrinidad** [comment]
Hi @benhillis,

OK!! I cleared the error.

I got it going by manually deleting all the files/folders under the 'lxss' folder structure: C:\Users\username\AppData\Local\lxss

I may have cause some kind of corruption as I'm loading stuff and pushing the limits on this subsystem. I'm finally rebuilding my Bash environment as I'm showcasing it at our Orlando MVP Community Event later this week.

Let me see how far can I get. 
:)

**Comment by MaximoTrinidad** [comment]
@benhillis,

It happens again! I got the "Uninstalling... Error: 0x80070091" on my third rebuild on Bash.
I will proceed to get collect the log information.
![bashunistall_2017-05-31_10-19-09](https://cloud.githubusercontent.com/assets/19892435/26636413/b855ade4-45ea-11e7-9c18-3947fe34f6ae.jpg)

**Comment by MaximoTrinidad** [comment]
@benhillis,

I ran the logman commands, try again to uninstall Bash again. It gave me the message:

C:\Users\mtrinidad>lxrun /uninstall /full
This will uninstall Ubuntu on Windows.
This will remove the Ubuntu environment as well as any modifications, new applications, and user data.
Type "y" to continue: y
Uninstalling...
The LX subsystem has an install or uninstall operation pending.

I attached all logs result.  
[lxcore_logs.zip](https://github.com/Microsoft/BashOnWindows/files/1041892/lxcore_logs.zip)

**Comment by brandon-kohn** [comment]
I have this issue as well.

**Comment by JadynWong** [comment]
I update 'Windows 10 Fall Creators Update'.I have this issue as well.

**Comment by kavdev** [comment]
```
C:\Users\x>lxrun /uninstall /full
Warning: lxrun.exe is only used to configure the legacy Windows Subsystem for Linux distribution.
Distributions can be installed by visiting the Windows Store:
https://aka.ms/wslstore

This will uninstall Ubuntu on Windows.
This will remove the Ubuntu environment as well as any modifications, new applications, and user data.
Type "y" to continue: y
Uninstalling...
The Windows Subsystem for Linux has an install or uninstall operation pending.
```

Same boat. I even deleted the lxss directory as suggested in another issue thread.

**Comment by riverar** [comment]
I believe this repros pretty easily:
- Install openSUSE and "launch" it
- Kill the console window that appears (while it's installing)
- Try starting bash

**Comment by bitcrazed** [comment]
Thanks for reporting this issue.

Alas, this issue exists in the beta WSL installation infrastructure (i.e. via `lxrun`), which has been deprecated and replaced by the FAR more reliable Store-based installation mechanism. Therefore, we recommend manually deleting the legacy distro folder from an elevated Console:

1. Close all open WSL Consoles
1. Open an elevated Cmd Console and execute: `del /s %localappdata%\lxss\`

Instead of installing and running the beta/legacy "Bash on Ubuntu on Windows" instance, we strongly recommend installing a distro of your choice from the Windows Store on an up to date version of Windows 10.

**Comment by riverar** [comment]
Am confused with the closure, this wasn't an `lxrun` install bug. It's an uninstall bug and is the only way to uninstall this stuff from Windows when things go wrong, no?

**Comment by bitcrazed** [comment]
To uninstall a Store-delivered distro, use the UI or PowerShell, e.g.:

``` PowerShell
Get-AppxPackage *Ubuntu* | Remove-AppxPackage
```

**Comment by ryandward** [comment]
The store doesn't have Arch...

**Comment by bitcrazed** [comment]
@ryandward Follow @tara-raj on Twitter [here](https://twitter.com/tara_msft): She'll be announcing new distros as and when they arrive :)

**Comment by llpaul** [comment]
I had very similar issues, and ultimately found running lxrun /install resolved it.

**Comment by bitcrazed** [comment]
@llpaul Please note that 

1. The "legacy distro" is now deprecated 
1. The "legacy distro" is installed via `lxrun …` (removed in Fall Creators Update), and uninstalled using `lxrun /uninstall ...`

Because the legacy distro is not supported, we strongly recommend you [install a distro from the Windows Store](ms-windows-store://collection/?CollectionId=LinuxDistros)

**Comment by dwumpus** [comment]
Windows Store is very often NOT an option in corporate america, by policy. We have a client that has the Windows Store disabled.  So, following manual instructions to install the product, but then not being able to uninstall it from the command-line or otherwise is definitely sub-optimal.  I will attempt to re-install using lxrun then try uninstalling since that seemed to work for others?  And will update if that does.

**Comment by dwumpus** [comment]
I was able to run lxrun /install and get it to work, just for those who were wondering.  And despite the Windows Store being disable, it reported it was 'downloading from the Windows Store'.  I believe this is because the Ubuntu distribution app was already manually installed into the local cache, and it was 'downloading it' from my local store.  I could be wrong.
